Sources/PackageLoading/CodableResult.swift

```diff
@@ -0,0 +1,64 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2019 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+import Basic
+import SPMLLBuild
+
+/// Codable wrapper for Result
+struct CodableResult<Success, Failure>: LLBuildValue where Success: Codable, Failure: Codable & Error {
```

Comment:
I think we should move this type to the `Basic` target since it'll be useful across the project (similar to CodableRange). We can conform this type to `LLBuildValue` via an extension.